internal/services/storage/storage_account_resource.go

```diff
@@ -2943,11 +2951,11 @@ func setEndpointAndHost(d *pluginsdk.ResourceData, ordinalString string, endpoin
 func getDefaultAllowBlobPublicAccess() bool {
 	// The default value for the field that controls if the blobs that belong to a storage account
 	// can allow anonymous access or not will change from false to true in 3.0.
-	return features.ThreePointOh()
+	return features.ThreePointOhBetaResources()
```

Comment:
this should be gated behind 3.0 not the new app service resources:

```suggestion
	return features.ThreePointOh()
```